Lib/test/test_clinic.py

```diff
@@ -249,6 +249,59 @@ def test_cpp_monitor_fail_no_matching_if(self):
         out = self.expect_failure(raw)
         self.assertEqual(out, msg)
 
+    def test_directive_output_unknown_preset(self):
+        out = self.expect_failure("""
+            /*[clinic input]
+            output preset nosuchpreset
+            [clinic start generated code]*/
+        """)
+        msg = "Unknown preset 'nosuchpreset'"
+        self.assertIn(msg, out)
+
+    def test_directive_output_cant_pop(self):
+        out = self.expect_failure("""
+            /*[clinic input]
+            output pop
+            [clinic start generated code]*/
+        """)
+        msg = "Can't 'output pop', stack is empty"
+        self.assertIn(msg, out)
+
+    def test_directive_output_print(self):
+        raw = dedent("""
+            /*[clinic input]
+            output print 'I told you once.'
+            [clinic start generated code]*/
+        """)
+        out = self.clinic.parse(raw)
+        # The generated output will differ for every run, but we can check that
+        # it starts with the clinic block, we check that it contains all the
+        # expected fields, and we check that it contains the checksum line.
+        self.assertTrue(out.startswith(dedent("""
+            /*[clinic input]
+            output print 'I told you once.'
+            [clinic start generated code]*/
+        """)))
+        fields = {
```

Comment:
Why `set`? :)
Since it is unordered it is not the best candidate for iteration in my opinion.

Reply:
Order is not important, so a set is fine. I agree it is a little bit surprising, though. A good argument for changing it to a tuple, is that a tuple would not generate confusion for future contributors :)

Reply:
Actually if the order shouldn't matter, a set is a good way to ensure it.

Reply:
True. Also, I'm on mobile, so I won't change it now anyways :)